.github/workflows/generate-pinval.yaml

```diff
@@ -149,57 +184,34 @@ jobs:
       - name: Install Hugo
         run: sudo dpkg -i ${{ runner.temp }}/hugo.deb
 
-      - name: Generate PINVAL reports (triad shard or explicit PINs)
-        run: |
-          set -euo pipefail
-
-          declare -a ARGS=(
-            --run-id "${{ inputs.run_id }}"
-            --triad "${{ inputs.triad_name }}"
-            --township "$TOWNSHIP"
-          )
-
-          # Convert comma-separated input into an array, stripping blanks
-          IFS=',' read -ra PINS_ARR <<< "${{ inputs.pins }}"
-          CLEAN_PINS=()
-          for p in "${PINS_ARR[@]:-}"; do
-            [[ -n "$p" ]] && CLEAN_PINS+=("$p")
-          done
-
-          if [[ ${#CLEAN_PINS[@]} -gt 0 ]]; then
-            ARGS+=(--pin "${CLEAN_PINS[@]}")
-          fi
-
-          echo "Running generate_pinval.py with: ${ARGS[*]}"
-          python3 scripts/generate_pinval/generate_pinval.py "${ARGS[@]}"
-
-      - name: Resolve target S3 bucket
-        id: bucket
+      - name: Generate PINVAL reports
+        env:
+          RUN_ID: ${{ needs.setup.outputs.run-id }}
         run: |
-          if [[ "${{ inputs.environment }}" == 'Development bucket' ]]; then
-            echo "bucket=${{ vars.AWS_S3_BUCKET_PINVAL_STAGING }}" >> "$GITHUB_OUTPUT"
-          else
-            echo "bucket=${{ vars.AWS_S3_BUCKET_PINVAL_PROD }}" >> "$GITHUB_OUTPUT"
-          fi
+          read -ra PINS <<< "${{ inputs.pins }}"
+          python3 scripts/generate_pinval/generate_pinval.py \
+            --run-id "$RUN_ID" \
+            --township "$TOWNSHIP" \
+            --pin "${PINS[@]}"
```

Comment:
If my understanding is correct, we now support `year` but it isn't built into this as a flag because the `resolve-metadata` script handles the linkage between the two and uses the corresponding `run_id`

Reply:
Exactly right! I thought about adding `--year` as an option for the `generate_pinval.py` script so that it could optionally infer the run ID in the same way that `resolve_metadata.py` does, but in the end I decided to go with the simpler design and leave it out. I think it might make sense to add `--year` at some point in the future if we find ourselves running `generate_pinval.py` frequently on the command line, but I expect we will mostly run it via the workflow, in which case the `--year` option is unnecessary.